src/WebJobs.Script.WebHost/Security/ScriptSecretSerializer.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public static class ScriptSecretSerializer
+    {
+        private static List<IScriptSecretSerializer> _secretFormatters = new List<IScriptSecretSerializer>
+        {
+            new ScriptSecretSerializerV0(),
+            new ScriptSecretSerializerV1()
+        };
+
+        private static IScriptSecretSerializer DefaultSerializer => GetSerializer(_secretFormatters.Max(i => i.SupportedFormatVersion));
+
+        public static string SerializeFunctionSecrets(IList<Key> secrets) => DefaultSerializer.SerializeFunctionSecrets(secrets);
+
+        public static string SerializeHostSecrets(HostSecrets secrets) => DefaultSerializer.SerializeHostSecrets(secrets);
+
+        public static IList<Key> DeserializeFunctionSecrets(string secretsJson) => ResolveSerializerAndRun(secretsJson, (s, o) => s.DeserializeFunctionSecrets(o));
+
+        public static HostSecrets DeserializeHostSecrets(string secretsJson) => ResolveSerializerAndRun(secretsJson, (s, o) => s.DeserializeHostSecrets(o));
+
+        private static TResult ResolveSerializerAndRun<TResult>(string secretsJson, Func<IScriptSecretSerializer, JObject, TResult> func)
+        {
+            JObject secrets = JObject.Parse(secretsJson);
```

Comment:
I know we probably don't have good error handling currently for this, but I'm wondering what happens if json parsing happens here? We should probably ensure we output a decent error message in such cases.